logstash-core/lib/logstash/environment.rb

```diff
@@ -3,8 +3,16 @@
 require "logstash/config/cpu_core_strategy"
 require "logstash/settings"
 require "socket"
+require "stud/temporary"
 
 module LogStash
+  # In the event that we're requiring this file without bootstrap/environment.rb
+  if !defined?(LogStash::Environment::LOGSTASH_HOME)
+    module Environment
+      LOGSTASH_HOME = Stud::Temporary.directory("logstash-home")
+      Dir.mkdir(::File.join(LOGSTASH_HOME, "data"))
```

Comment:
+1 to using `FileUtils.mkdir_p`.  If the path is deeply nested (and `/usr/share/logstash/data` certainly fits that bill), then this is a safer approach.